cmake/Modules/FindGoogleSanitizer.cmake

```
@@ -0,0 +1,97 @@
+# Find google sanitizers
+#
+# This module sets the following targets: GoogleSanitizer::address
+# GoogleSanitizer::thread GoogleSanitizer::undefined GoogleSanitizer::leak
+# GoogleSanitizer::memory
+include_guard(GLOBAL)
+
+option(NO_SANITIZE_VPTR "no vptr sanitizer" OFF)
+
+get_property(languages GLOBAL PROPERTY ENABLED_LANGUAGES)
+
+set(_source_code
+    [==[
+  #include <stdio.h>
+  int main() {
+  printf("hello world!");
+  return 0;
+  }
```

Would't it be better to use something like the following as the test program and then validate that if run with ASAN without arguments it passes, but if run with -1 or say 17 it fails?
```suggestion
  #include <stdlib.h>
  int main(int argc, char *argv[]) {
  int idx = argc > 1 ? atoi(argv) : 1;
  int return_codes[] = {0, 1, 2};
  return return_codes[idx];
  return 0;
  }
```

---

@melfet I have to revert your commit because the main purpose is to provide a simple correct program to  test link flags. If instead we give  a problematic code, check_source_runs will return error